apps/condo/domains/ticket/components/TicketClassifierSelect.tsx

```diff
@@ -0,0 +1,320 @@
+    } = useTicketClassifierSelectHook({
+        onChange: (id) => onUserSelect(id, TicketClassifierTypes.category),
+        onSearch: (id) => onUserSearch(id, TicketClassifierTypes.category),
+        initialValue: categoryClassifier,
+    })
+
+    const {
+        set: placeSet,
+        SelectComponent: PlaceSelect,
+        ref: placeRef,
+    } = useTicketClassifierSelectHook({
+        onChange: (id) => onUserSelect(id, TicketClassifierTypes.place),
+        onSearch: (id) => onUserSearch(id, TicketClassifierTypes.place),
+        initialValue: placeClassifier,
+    })
+
+    const Setter = {
+        place: placeSet,
+        category: categorySet,
+        description: descriptionSet,
+    }
+
+    useEffect(() => {
+        helper.init().then(_ => {
+            if (threeLvlSelectState.current.id) {
+                helper.findRules({ id: threeLvlSelectState.current.id }).then(([rule]) => {
+                    const { place, category, description } = rule
+                    threeLvlSelectState.current = { ...threeLvlSelectState.current, ...{ place: place.id, category: category.id, description: description.id } }
+                    updateLevels(threeLvlSelectState.current)
+                })
+            } else {
+                helper.search('', 'place').then(places => {
+                    Setter.place.all(places)
+                })
+                helper.search('', 'category').then(categories => {
+                    Setter.category.all(categories)
+                })
+            }
+        })
+        return () => {
+            // clear all loaded data from helper
+            helper.clear()
+        }
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+    }, [])
+
+    const load3levels = async () => {
+        const { place, category, description } = threeLvlSelectState.current
+        console.log('load3levels: >>> threeLvlSelectState.current', threeLvlSelectState.current)
+        const loadedRules = await Promise.all([
+            { category, description, type: 'place' },
+            { place, description, type: 'category' },
+            { place, category, type: 'description' },
+        ].map(selector => {
+            const { type, ...querySelectors } = selector
+            return new Promise<[string, Options[]]>(resolve => {
+                const query = {}
+                for (const key in querySelectors) {
+                    if (querySelectors[key]) {
+                        query[key] = { id: querySelectors[key] }
+                    }
+                }
+                console.log('load3levels - query: ', query)
+                helper.findRules(query).then(data => resolve([type, helper.rulesToOptions(data, type)]))
+            })
+        }))
+        const result = Object.fromEntries(loadedRules)
+        return result
+    }
+
+    const openSelect = (ref) => {
+        if (ref.current) {
+            ref.current.blur()
+            setTimeout(ref.current.focus, 0)
```

Comment:
Why we couldn't use default aunt Select API for this?

Reply:
we need to set the focus for text typing and open select in some cases

Default Ant'd API allows only to change the way select opens -> on focus = I use this to open it on manual focus call
and to set the default open - Initial open state of dropdown - we don't need to open it on init
the open property does not set a focus for typing